sdk/tests/wallet/core.rs

```diff
@@ -99,11 +100,7 @@ async fn changed_bip_path() -> Result<()> {
 
     // Building the wallet with another coin type needs to return an error, because a different coin type was used in
     // the existing account
-    let mismatch_err: Result<Wallet> = Err(Error::BipPathMismatch {
-        new_bip_path: Some(Bip44::new(IOTA_COIN_TYPE)),
-        old_bip_path: Some(Bip44::new(SHIMMER_COIN_TYPE)),
-    });
-    assert!(matches!(err, mismatch_err));
+    assert!(matches!(result, Err(Error::BipPathMismatch { .. })));
```

Comment:
Why make the error less precise? If that's supposed to be fixed later in the TODO, I wouldn't actually change this since it won't fail anyway. Feels like we could forget to revert to a more precise error later.